src/kakarot/instructions/system_operations.cairo

```diff
@@ -584,12 +585,16 @@ namespace CallHelper {
 
         let stack = Stack.push(ctx.stack, status);
         let ctx = ExecutionContext.update_stack(ctx, stack);
+        // in the case that the return data is empty, we use `slice_data` 
+        // to give us a zero padded array
+        let return_data_len = ctx.sub_context.return_data_len;
+        let return_data = Helpers.slice_data(return_data_len, ctx.sub_context.return_data, 0, return_data_len);
```

Comment:
how can this solve the case where `return_data_len` is not consistent with `return_data` anymore?